packages/@aws-cdk/aws-appmesh/lib/gateway-route-spec.ts

```diff
@@ -7,26 +12,232 @@ import { IVirtualService } from './virtual-service';
+    return new GatewayRouteHostnameMatchImpl({
+      exact: name,
+    });
+  }
+
+  /**
+   * The value of the host name with the given name must end with the specified characters.
+   *
+   * @param suffix The specified ending characters of the host name to match on
+   */
+  public static endingWith(suffix: string): GatewayRouteHostnameMatch {
+    return new GatewayRouteHostnameMatchImpl({
+      suffix: suffix,
+    });
+  }
+
+  /**
+   * Returns the gateway route host name match configuration.
+   */
+  public abstract bind(scope: Construct): GatewayRouteHostnameMatchConfig;
+}
+
+class GatewayRouteHostnameMatchImpl extends GatewayRouteHostnameMatch {
+  constructor(
+    private readonly matchProperty: CfnGatewayRoute.GatewayRouteHostnameMatchProperty,
+  ) { super(); }
+
+  bind(_scope: Construct): GatewayRouteHostnameMatchConfig {
+    return {
+      hostnameMatch: this.matchProperty,
+    };
+  }
+}
+
+/**
+ * The criterion for determining a request match for this GatewayRoute.
  */
 export interface HttpGatewayRouteMatch {
   /**
-   * Specifies the path to match requests with.
-   * This parameter must always start with /, which by itself matches all requests to the virtual service name.
-   * You can also match for path-based routing of requests. For example, if your virtual service name is my-service.local
-   * and you want the route to match requests to my-service.local/metrics, your prefix should be /metrics.
+   * Specify how to match requests based on the 'path' part of their URL.
+   *
+   * @default - matches requests with any path
+   */
+  readonly path?: HttpRoutePathMatch;
+
+  /**
+   * Specifies the client request headers to match on. All specified headers
+   * must match for the gateway route to match.
+   *
+   * @default - do not match on headers
+   */
+  readonly headers?: HeaderMatch[];
+
+  /**
+   * The gateway route host name to be matched on.
+   *
+   * @default - do not match on host name
+   */
+  readonly hostname?: GatewayRouteHostnameMatch;
+
+  /**
+   * The method to match on.
+   *
+   * @default - do not match on method
+   */
+  readonly method?: HttpRouteMethod;
+
+  /**
+   * The query parameters to match on.
+   * All specified query parameters must match for the route to match.
+   *
+   * @default - do not match on query parameters
    */
-  readonly prefixPath: string;
+  readonly queryParameters?: QueryParameterMatch[];
+
+  /**
+   * When `true`, rewrites the original request received at the Virtual Gateway to the destination Virtual Service name.
+   * When `false`, retains the original hostname from the request.
+   *
+   * @default true
+   */
+  readonly rewriteRequestHostname?: boolean;
+
+  /**
+   * The in-coming request's path to be rewritten when the request is matched.
+   *
+   * @default - the matched part of the path (which is all of it by default) is rewritten to '/'
+   */
+  readonly pathRewrite?: HttpGatewayRoutePathRewrite;
 }
```

Comment:
The PR is missing updating the documentation of the match property in `HttpGatewayRouteSpecOptions`.

Reply:
Addressed.